File: my_project_typescript/medium_problems_typescript/from51to100/rotateArray.ts
/**
 Do not return anything, modify nums in-place instead.
 */
 function rotate(nums: number[], k: number): void {
    k=k % nums.length;
    for (let i=0; i< k; i++) {
     
     const lastItem: number = nums.pop();
     nums.unshift(lastItem);
    }
};

File: src/my_project/interviews/top_150_questions_round_22/rotate_array.py
from typing import List, Union, Collection, Mapping, Optional
from abc import ABC, abstractmethod

class Solution:
    def rotate(self, nums: List[int], k: int) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """

        len_nums = len(nums)
        k = k % len_nums

        nums[:] = nums[len_nums-k:] + nums[:len_nums-k]

        return nums
